tokio/src/blocking.rs

```diff
@@ -45,16 +48,16 @@ cfg_not_rt! {
         type Output = Result<R, std::io::Error>;
 
         fn poll(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<Self::Output> {
-            unreachable!()
+            match self.never {}
         }
     }
 
     impl<T> fmt::Debug for JoinHandle<T>
     where
         T: fmt::Debug,
     {
-        fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
-            fmt.debug_struct("JoinHandle").finish()
+        fn fmt(&self, _fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+            match self.never {}
```

Comment:
Since I think we just need a few comments, we don't need a new `Never` type. Thus I prefer to revert this change.